### ISSUE TITLE
Feature request: Idle sleep activation, no timers

### ISSUE THREAD

**Issue by makedir** [lukaslangrock/ShutdownTimerClassic#18]
Im looking since years for a opensource program for Windows like this tool, where you can trust the code and wont have to worry that maybe it contains malicious code, which offers the functionality like your tool but also keeps the laptop entering standby / execute command after n time and m no mouse movements.

Could you please implement these simple features:

- timer which resets when you move the mouse
- after timer, system executes command, you could use internal command like standby/shutdown/hibernate, or let the user configure a executable or bat file

**Comment by lukaslangrock** [comment]
Hi there and thanks for your feedback.

I am curious as to why you'd want a feature where the timer resets when there is mouse movement detected. I currently don't really see how that would be a feature of interest (to many).
As to the second feature, this one has been on my mind as well and I am working on implementing that.

**Comment by makedir** [comment]
A timer is important and a needed feature to make a program like this work to put a laptop to sleep, as simple as that. The inbuild idle timer of Windows is not reliable, because it can be deactivated by lots of programs (watching a movie, having a video player open, and so on). ERGO you need a 3rd party program with a timer, which is just bound to mouse movements. That is the only reliable way. If you let your laptop on with a video player for example, it will burn in the monitor and also stay on forever.

Please implement both features, they are important. And simple to implement. Thank you.

**Comment by lukaslangrock** [comment]
I see now, that would indeed be a good feature. I'll work on that one too.

**Comment by makedir** [comment]
Thank you that would be great. Maybe you could also implement the 3 main power modes of Windows too next to standby, hibernate, S3 and modern standby as a command to run, or you give a command argument which will be executed so people can select a bat file or ahk (autohotkey) script to do that.

**Comment by makedir** [comment]
Yes it does indeed 😅sorry that is redundant then. Ive not looked at the program so far, because I need the mouse timer for using it.

**Comment by lukaslangrock** [comment]
No problem haha, it'll not be long until the update :)

**Comment by DouglasMoody** [comment]
Status update?

**Comment by lukaslangrock** [comment]
The feature for executing custom scripts/commands was implemented, the feature regarding resetting the timer on mouse movement is planned but not yet worked on.
Both features will probably ship in the next update.

**Comment by MatheusOliveira-dev** [comment]
I was analyzing and attempting to implement the requested functionality of globally capturing mouse movements, particularly when the application is minimized, and encountered the following complications:

The functionality of globally capturing mouse movements, especially when the application is minimized, is not fully achievable due to various implications related to user privacy and Windows operating system security:

- **User Privacy**: Windows implements security measures to protect user privacy. This means that certain system events, such as mouse movements, can only be intercepted when the application registering the hook is active and in the foreground. This prevents applications from accessing user information without explicit knowledge or consent.

- **Abuse Prevention**: The Windows operating system disables low-level hooks when the registering application is minimized or in the background. This measure is taken to prevent abuse by malicious applications that may attempt to capture mouse events without user consent. Thus, the system safeguards user security and privacy of data.

- **Security Restrictions**: Windows enforces security restrictions to protect the integrity and stability of the operating system. These restrictions ensure that applications can only access necessary resources in a controlled manner, limiting the ability to intercept system events when the application is not actively interacting with the user.



@DouglasMoody @makedir: Does it make sense to implement this feature only when the 'Run in background' option is not used? With the countdown running without this option, the application is not minimized, so it's possible to monitor mouse movements.


@lukaslangrock, are you already working on this? I haven't found a native way to do this without using third-party packages.

**Comment by lukaslangrock** [comment]
I haven't researched the functionality / Windows-aspect, but I started working on preparing the UI and logic for this feature, keeping hotkey support in mind too. My approach was based on this [NuGet package](https://www.nuget.org/packages/MouseKeyHook), but I have not tested it yet.

**Comment by lukaslangrock** [comment]
I uploaded my progress in a new [branch](https://github.com/lukaslangrock/ShutdownTimerClassic/tree/feature-mouse-keyboard-hooks). Using this NuGet package it is not possible to detect movements during background operation, but as long as the countdown window is visible (doesn't even need to be in the foreground or focussed), the timer will reset.
The feature still needs some work, tho. This is just a demo.
